datafusion/expr-common/src/signature.rs

```diff
@@ -460,6 +521,44 @@ fn get_data_types(native_type: &NativeType) -> Vec<DataType> {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialOrd)]
+pub struct Coercion {
+    pub desired_type: TypeSignatureClass,
+    pub allowed_casts: Vec<TypeSignatureClass>,
```

Comment:
What is `allowed_casts`s? Does that represent the source types that this coercion applies to?

If so, perhaps we could name this filed `allowed_source_types`